finality-aleph/src/sync/service.rs

```diff
@@ -313,6 +313,28 @@ where
         }
     }
 
+    fn handle_internal_request(&mut self, id: BlockIdFor<J>) {
+        trace!(
+            target: LOG_TARGET,
+            "Handling an internal request for block {:?}.",
+            id,
+        );
+        match self.handler.handle_internal_request(&id) {
+            Ok(true) => {
+                self.handle_task(RequestTask::new_block(id));
```

Comment:
Hm, currently the highest justified tasks are first added to the queue (sometimes with a zero delay), and only then handled. This approach might actually be better, but it would be nice to have consistency – either change `request_highest_justified` or make this put stuff in the queue.

Also, having this as a separate method might be good, as it would have made this inconsistency clearer – if you put it right next to `request_highest_justified` as `request_block` you would have immediately noticed that the bodies don't quite fit. :P